conjure-java-core/src/test/java/com/palantir/conjure/java/types/EndpointErrorGeneratorTests.java

```diff
@@ -0,0 +1,89 @@
+import com.palantir.conjure.spec.EndpointError;
+import com.palantir.conjure.spec.EndpointName;
+import com.palantir.conjure.spec.ErrorCode;
+import com.palantir.conjure.spec.ErrorDefinition;
+import com.palantir.conjure.spec.ErrorNamespace;
+import com.palantir.conjure.spec.ErrorTypeName;
+import com.palantir.conjure.spec.HttpMethod;
+import com.palantir.conjure.spec.HttpPath;
+import com.palantir.conjure.spec.ServiceDefinition;
+import com.palantir.conjure.spec.TypeName;
+import com.palantir.logsafe.exceptions.SafeIllegalStateException;
+import java.util.List;
+import org.junit.jupiter.api.Test;
+
+public final class EndpointErrorGeneratorTests {
+    private static final String ERROR_NAME = "TestError";
+    private static final ConjureDefinition DEFINITION = ConjureDefinition.builder()
+            .version(1)
+            .errors(List.of(ErrorDefinition.builder()
+                    .errorName(TypeName.of(ERROR_NAME, "com.palantir.test"))
+                    .code(ErrorCode.CUSTOM_SERVER)
+                    .namespace(ErrorNamespace.of("TestService"))
+                    .build()))
+            .services(List.of(ServiceDefinition.builder()
+                    .serviceName(TypeName.of("TestService", "com.palantir.test"))
+                    .endpoints(List.of(EndpointDefinition.builder()
+                            .endpointName(EndpointName.of("testEndpoint"))
+                            .httpMethod(HttpMethod.GET)
+                            .httpPath(HttpPath.of("/test"))
+                            .errors(EndpointError.builder()
+                                    .error(ErrorTypeName.builder()
+                                            .name(ERROR_NAME)
+                                            .package_("com.palantir.test")
+                                            .namespace(ErrorNamespace.of("TestService"))
+                                            .build())
+                                    .build())
+                            .build()))
+                    .build()))
+            .build();
+
+    @Test
+    public void testThrowsExceptionWhenOptionIsFalseAndEndpointErrorsExist() {
+        // Create an EndpointErrorGenerator with the option set to false
+        EndpointErrorGenerator generator = new EndpointErrorGenerator(Options.builder()
+                .dangerousDoNotUseEnableEndpointAssociatedErrors(false)
+                .build());
+
+        // Verify that an exception is thrown, and that it contains the name of the error associated with the endpoint
+        assertThatThrownBy(() -> generator.generate(DEFINITION))
+                .isInstanceOf(SafeIllegalStateException.class)
+                .hasMessageContaining("Errors are associated with endpoints. This feature is currently not supported.")
+                .hasMessageContaining(ERROR_NAME);
+    }
+
+    @Test
+    public void testNoExceptionThrownWhenOptionIsTrueAndEndpointErrorsExist() {
```

Comment:
nit: we should also have a test that no exception is thrown if option is false, but there are no endpoint errors